### PR TITLE
python tutorial

### PR DESCRIPTION
**Pull request by PaLeroy**

### CODE REVIEW

File: python_tutorial/README.md [hunk 1]
```diff
@@ -0,0 +1,1017 @@
+# Python tutorial
+In the different projects, you will be asked to code using Python 3.6.
+
+In this tutorial, you will install Python and learn its basics. This tutorial is strongly based on the [**Berkley's tutorial**](https://inst.eecs.berkeley.edu/~cs188/su19/project0/).
```
Comment (glouppe) [inline python_tutorial/README.md]: -> 'based on CS188 project 0 from UC Berkeley'

File: python_tutorial/README.md [hunk 2]
```diff
@@ -0,0 +1,1017 @@
...
+    - [More References](#more-references)
+- [Exercises](#exercises)
+    - [Question 1: Addition](#question-1-addition)
+    - [Question 2: buyLotsOfFruit function](#question-2-buylotsoffruit-function)
+    - [Question 3: shopSmart function](#question-3-shopsmart-function)
+
+## Tools
+In order to work with Python, we encourage you to use a text editor and a terminal. Make sure to read the whole tools section before starting your installation.
+
+On Unix and OS X, you can use the basic terminal while on windows, you will use the *Anaconda prompt* (see further).
```
Comment (glouppe) [inline python_tutorial/README.md]: windos -> Windows

File: python_tutorial/README.md [hunk 3]
```diff
@@ -0,0 +1,1017 @@
...
+
+There are plenty of text edtior that can be used but [Atom](https://atom.io/) or [Sublime Text](https://www.sublimetext.com/) are good candidates.
```
Comment (glouppe) [inline python_tutorial/README.md]: edtior -> editors

File: python_tutorial/README.md [hunk 4]
```diff
@@ -0,0 +1,1017 @@
...
+
+Also, there exist many IDEs such as [Spyder](https://www.spyder-ide.org/) or [PyCharm](https://www.jetbrains.com/pycharm/) that could be used **CAREFULLY**.
+
+### Python
+We recommand you to install python
```
Comment (glouppe) [inline python_tutorial/README.md]: recommand -> recommend
python -> Python

File: python_tutorial/README.md [hunk 5]
```diff
@@ -0,0 +1,1017 @@
...
+
+On Unix and OS X, you can use the basic terminal while on windows, you will use the *Anaconda prompt* (see further).
+
+There are plenty of text edtior that can be used but [Atom](https://atom.io/) or [Sublime Text](https://www.sublimetext.com/) are good candidates.
+
+Also, there exist many IDEs such as [Spyder](https://www.spyder-ide.org/) or [PyCharm](https://www.jetbrains.com/pycharm/) that could be used **CAREFULLY**.
+
+### Python
+We recommand you to install python 
+[Conda](https://docs.anaconda.com/anaconda/install/). It allows you to manage many environments, with different Python versions and dependencies.
```
Comment (glouppe) [inline python_tutorial/README.md]: Conda -> Anaconda

File: python_tutorial/README.md [hunk 6]
```diff
@@ -0,0 +1,1017 @@
...
+
+For windows users, Conda comes with the *Anaconda prompt* that will be used as a terminal to execute Python code.
```
Comment (glouppe) [inline python_tutorial/README.md]: windows -> Windows
Conda -> Anaconda

File: python_tutorial/README.md [hunk 7]
```diff
@@ -0,0 +1,1017 @@
...
+
+### Unix & OS X basics
+When you open a terminal window, you’re placed at a command prompt:
+
+```
```
Comment (glouppe) [inline python_tutorial/README.md]: Use "```console" to turn on highlighting

Comment (glouppe) [reply python_tutorial/README.md]: (here and everywhere below)

File: python_tutorial/README.md [hunk 8]
```diff
@@ -0,0 +1,1017 @@
...
+MacBook-Pro-de-Pascal:foo pascalleroy$ ls
+MacBook-Pro-de-Pascal:foo pascalleroy$ touch hello
+MacBook-Pro-de-Pascal:foo pascalleroy$ ls
+hello
+MacBook-Pro-de-Pascal:foo pascalleroy$ cd ..
+MacBook-Pro-de-Pascal:~ pascalleroy$ 
+```
+
+Download **[LINK TO ZIP]()** into a directory and change your terminal to that directory(note: the zip file’s name may be slightly different when you download it). Use `unzip` to extract the contents of the zip file:
```
Comment (glouppe) [inline python_tutorial/README.md]: change this

File: python_tutorial/README.md [hunk 9]
```diff
@@ -0,0 +1,1017 @@
...
+- `rm` removes (deletes) a file
+- `mv` moves a file (i.e., cut/paste instead of copy/paste)
+- `man` displays documentation for a command
+- `pwd` prints your current path
+- Press “Ctrl-c” to kill a running process
+- Append `&` to a command to run it in the background
+- `fg` brings a program running in the background to the foreground
+
+### Windows basics
+Since you are on Windows, you will use the *Anaconda prompt* that is installed with Conda. In this terminal, the commands are different from those of UNIX.
```
Comment (glouppe) [inline python_tutorial/README.md]: Conda -> Anaconda

File: python_tutorial/README.md [hunk 10]
```diff
@@ -0,0 +1,1017 @@
...
+
+Download **[LINK TO ZIP]()** into a directory of your choice, unzip it using your file explorer and change the current directory of the *Anaconda prompt* to this new directory.
```
Comment (glouppe) [inline python_tutorial/README.md]: update link

File: python_tutorial/README.md [hunk 11]
```diff
@@ -0,0 +1,1017 @@
...
+In the following, the commands are the same either you are working on Anaconda prompt or a UNIX terminal.
+
+#### Creating a Conda Environment
+The command for creating a conda environment with Python 3.6 is:
+
+```
+conda create --name <env-name> python=3.6
+```
+
+For us, we decide to name our environment cs188, so we run the following command, and press y to confirm installing any missing packages.
```
Comment (glouppe) [inline python_tutorial/README.md]: cs188 -> info8006

File: python_tutorial/README.md [hunk 12]
```diff
@@ -0,0 +1,1017 @@
...
+
+#### Leaving the Environment
+Leaving the environment is just as easy.
+```
+(info8006) MacBook-Pro-de-Pascal:python_basics pascalleroy$ conda deactivate
+MacBook-Pro-de-Pascal:python_basics pascalleroy$ python -V
+Python 2.7.10
+```
+
+Our python version has now returned to whatever the system default is!
```
Comment (glouppe) [inline python_tutorial/README.md]: python -> Python

File: python_tutorial/README.md [hunk 13]
```diff
@@ -0,0 +1,1017 @@
...
+
+## Python Basics
+### Required Files
+Following the terminal basics of your OS, you have normally dowloaded a zip file, extracted the content of this archive in a chosen directory and the current directoty of your terminal should contains those files. 
+
+### Introduction
+
+The programming assignments in this course will be written in Python, an interpreted, object-oriented language that shares some features with both Java and Scheme. This tutorial will walk through the primary syntactic constructions in Python, using short examples.
+
+We encourage you to type all python shown in the tutorial onto your own machine. Make sure it responds the same way.
```
Comment (glouppe) [inline python_tutorial/README.md]: We encourage you to type in yourself the Python code of this tutorial. Make sure it produces the same results.

File: python_tutorial/README.md [hunk 14]
```diff
@@ -0,0 +1,1017 @@
...
+
+You may find the Troubleshooting section helpful if you run into problems. It contains a list of the frequent problems previous CS188 students have encountered when following this tutorial.
```
Comment (glouppe) [inline python_tutorial/README.md]: add link to the Troubleshooting section

Comment (glouppe) [reply python_tutorial/README.md]: remove 'CS188'

File: python_tutorial/README.md [hunk 15]
```diff
@@ -0,0 +1,1017 @@
...
+[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
+Type "help", "copyright", "credits" or "license" for more information.
+>>> 
+```
+
+### Operators
+
+The Python interpreter can be used to evaluate expressions, for example simple arithmetic expressions. If you enter such expressions at the prompt (`>>>`) they will be evaluated and the result will be returned on the next line.
+
+```
```
Comment (glouppe) [inline python_tutorial/README.md]: Use "```python"

Comment (glouppe) [reply python_tutorial/README.md]: (here and everywhere below)

File: python_tutorial/README.md [hunk 16]
```diff
@@ -0,0 +1,1017 @@
...
+(info8006) MacBook-Pro-de-Pascal:python_basics pascalleroy$ python fruit.py
+That'll be 4.800000 please
+Sorry we don't have coconuts
+```
+
+### Advanced Exercises
+
+Write a `mean`function in Python. You can find two solutions in `mean.py`.
```
Comment (glouppe) [inline python_tutorial/README.md]: space between `mean` and 'function'

File: python_tutorial/README.md [hunk 17]
```diff
@@ -0,0 +1,1017 @@
...
+Welcome to the Berkeley Bowl fruit shop
+1.0
+Apples cost $1.00 at the Berkeley Bowl.
+Welcome to the Stanford Mall fruit shop
+4.5
+Apples cost $4.50 at the Stanford Mall.
+My, that's expensive!
+```
+
+So what just happended? The `import shop` statement told Python to load all of the functions and classes in `shop.py`. The `line berkeleyShop = shop.FruitShop(shopName, fruitPrices)` constructs an instance of the `FruitShop` class defined in `shop.py`, by calling the `__init__` function in that class. Note that we only passed two arguments in, while `__init__` seems to take three arguments: `(self, name, fruitPrices)`. The reason for this is that all methods in a class have `self` as the first argument. The `self` variable’s value is automatically set to the object itself; when calling a method, you only supply the remaining arguments. The `self` variable contains all the data (`name` and `fruitPrices`) for the current specific instance (similar to `this` in Java). The print statements use the substitution operator (described in the [Python docs](https://docs.python.org/2/library/stdtypes.html#string-formatting) if you’re curious).
```
Comment (glouppe) [inline python_tutorial/README.md]: refer to Python 3 documentation instead of Python 2

**Comment by glouppe** [comment]
@AWehenkel can you also have a look?

File: python_tutorial/README.md (hunk 8 above)
Comment (PaLeroy) [reply python_tutorial/README.md]: "Download **[LINK TO ZIP]()** into a directory of your choice. On your terminal, change the current directory to that chosen directory. Use `unzip` to extract the contents of the zip file:"

File: python_tutorial/README.md [hunk 18]
```diff
@@ -0,0 +1,1016 @@
...
+Welcome to the Berkeley Bowl fruit shop
+1.0
+Apples cost $1.00 at the Berkeley Bowl.
+Welcome to the Stanford Mall fruit shop
+4.5
+Apples cost $4.50 at the Stanford Mall.
+My, that's expensive!
+```
+
+So what just happended? The `import shop` statement told Python to load all of the functions and classes in `shop.py`. The `line berkeleyShop = shop.FruitShop(shopName, fruitPrices)` constructs an instance of the `FruitShop` class defined in `shop.py`, by calling the `__init__` function in that class. Note that we only passed two arguments in, while `__init__` seems to take three arguments: `(self, name, fruitPrices)`. The reason for this is that all methods in a class have `self` as the first argument. The `self` variable’s value is automatically set to the object itself; when calling a method, you only supply the remaining arguments. The `self` variable contains all the data (`name` and `fruitPrices`) for the current specific instance (similar to `this` in Java). The print statements use the substitution operator (described in the [Python docs](https://docs.python.org/3.6/) if you’re curious).
```
Comment (AWehenkel) [inline python_tutorial/README.md]: happened

**Comment by glouppe** [comment]
Thanks! I will merge this for now.

**Pull request merged**